dx: add CONTRIBUTING.md file

jideabdqudus opened the pull request:
Closes #71 

- Created and wrote the Contributing Doc to assist with developers experience.

ykdojo commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,84 @@
+# Contributing
+
+Defaang lives through contributions from the community, In contributing to this repository, please first discuss the change you wish to make via issue, discord, or any other method with the owners of this repository before making a change.
```
Defaang -> defaang

first commma -> period

via issue -> via an issue

owners -> maintainers

jideabdqudus replied:
Done!

ykdojo reviewed (changes requested):
This is a good start, but it's a lot to review in one go.

Any chance you could break it down to much smaller pieces and send individual PRs for them?

jideabdqudus commented:
> This is a good start, but it's a lot to review in one go.
> 
> Any chance you could break it down to much smaller pieces and send individual PRs for them?

Do I close this PR and send new PRs as individual chunks? Or how do I go about this

rohitdasu commented:
> > This is a good start, but it's a lot to review in one go.
> > Any chance you could break it down to much smaller pieces and send individual PRs for them?
> 
> Do I close this PR and send new PRs as individual chunks? Or how do I go about this

Hi @jideabdqudus you don't need to close this PR. Update the PR whichever changes requested by @ykdojo

subhoghoshX commented:
LGTM. Make the changes that YK mentioned and make sure to run it through a grammar checker like Grammarly or Quillbot.

jideabdqudus commented:
> LGTM. Make the changes that YK mentioned and make sure to run it through a grammar checker like Grammarly or Quillbot.

Changes have been made & I did the spell check as well cc @rohitdasu @ykdojo

subhoghoshX commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,84 @@
+
+## How to Help
+
+If you have some experiences with developing, we would gladly welcome your help in fixing issues and adding features. Even if you just start: There are tons of smaller things and chores to do, so by cleaning up code here and there you can improve your skills and also help the project. Even if it's a minor fix, it is still a fix!
```
experiences -> experience

jideabdqudus replied:
Fixed :)

ykdojo commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,84 @@
+
+## Where to Get Help
+
+An active community forms the backbone of every major Open Source project. The community is present on various platforms on the internet, where we actively help each other out – not only with regard to using the application. GitHub is the most tech-focused part of the community, albeit [discussing here](https://github.com/ykdojo/defaang/discussions) is the way to go for more theoretical discussio ...
```
I think this paragraph is too complex. Can you simplify?

ykdojo commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,84 @@
+
+## How to Help
+
+If you have some experience with developing, we would gladly welcome your help in fixing issues and adding features. Even if you just start: There are tons of smaller things and chores to do, so by cleaning up code here and there you can improve your skills and also help the project. Even if it's a minor fix, it is still a fix!
```
I really don't like the word chores here - I really don't think we have any *chores* to do? I think every task is interesting in a different way

ykdojo commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,84 @@
+
+### Issue Labels
+
+There are – broadly speaking – three categories of issues on the defaang GitHub issue tracker, which you can distinguish with their associated labels.
```
simplify this sentence

ykdojo commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,84 @@
+2. If you want to tackle an issue, but along the way realize that you ran out of time and can't finish it: _communicate it_! This is an Open Source project, and retracting is no shame, but not communicating that you cannot finish something will hinder all other people involved.
+3. If you have a PR and are not completely sure about whether it's right, or not, just open the PR and outline your questions there. It is much easier to work with something we can see and improve the PR than if you ask without something concrete.
+
+In any case, just remember to just open PRs and dive in. The worst that could happen is that we ask you to rewrite some file, but that has never happened and is pretty unlikely to happen in the future.
```
say "just" once here

ykdojo commented on CONTRIBUTING.md:
````diff
@@ -0,0 +1,84 @@
+
+#### Commit Example
+
+```bash
````
I don't think this should be rendered as a bash command

jideabdqudus replied:
I've made the new changes across the PR

ykdojo commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,82 @@
+# Contributing
+
+defaang lives through contributions from the community. In contributing to this repository, please first discuss the change you wish to make via an issue, discord, or any other method with the maintainers of this repository before making a change.
```
In contributing -> When contributing

ykdojo commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,82 @@
+
+**Table of Contents**
+
+- [Contributing](#contributing)
```
do these links work?

jideabdqudus replied:
Yeah

ykdojo commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,82 @@
+If you want to help, make sure to _communicate your intent_, that is:
+
+1. If you want to tackle an issue, comment and say that you want to develop a PR. In the PR, please link the issue.
+2. If you want to tackle an issue, but along the way realize that you ran out of time and can't finish it: _communicate it_! This is an Open Source project, and retracting is no shame, but not communicating that you cannot finish something will hinder all other people involved.
```
If you want to tackle an issue -> If you start working on an issue

ykdojo commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,82 @@
+2. If you want to tackle an issue, but along the way realize that you ran out of time and can't finish it: _communicate it_! This is an Open Source project, and retracting is no shame, but not communicating that you cannot finish something will hinder all other people involved.
+3. If you have a PR and are not completely sure about whether it's right, or not, just open the PR and outline your questions there. It is much easier to work with something we can see and improve the PR than if you ask without something concrete.
+
+In any case, just remember to just open PRs and dive in. The worst that could happen is that we ask you to rewrite some file, but that has never happened and is pretty unlikely to happen in the future.
```
I'd remove this part:

> The worst that could happen is that we ask you to rewrite some file, but that has never happened and is pretty unlikely to happen in the future.

ykdojo commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,82 @@
+
+## Remarks
+
+If something is missing here, or you feel something is not well described, either directly open a PR or ask one of the community places.
```
ask *in* one of the...

ykdojo commented on CONTRIBUTING.md:
```diff
@@ -50,11 +50,11 @@ There are three categories of issues on the defaang GitHub issue tracker:
 
-1. If you want to tackle an issue, comment and say that you want to develop a PR. In the PR, please link the issue.
-2. If you want to tackle an issue, but along the way realize that you ran out of time and can't finish it: _communicate it_! This is an Open Source project, and retracting is no shame, but not communicating that you cannot finish something will hinder all other people involved.
+1. If you want to start working on an issue, comment and say that you want to develop a PR. In the PR, please link the issue.
```
I think *tackle* here is ok? I just thought too much repeating won't sound good

jideabdqudus replied:
Okay, that makes sense.

ykdojo commented on CONTRIBUTING.md:
```diff
@@ -50,11 +50,11 @@ There are three categories of issues on the defaang GitHub issue tracker:
-1. If you want to tackle an issue, comment and say that you want to develop a PR. In the PR, please link the issue.
-2. If you want to tackle an issue, but along the way realize that you ran out of time and can't finish it: _communicate it_! This is an Open Source project, and retracting is no shame, but not communicating that you cannot finish something will hinder all other people involved.
+1. If you want to start working on an issue, comment and say that you want to develop a PR. In the PR, please link the issue.
+2. If you start working on an issue, but along the way realize that you ran out of time and can't finish it: _communicate it_! This is an Open Source project, and retracting is no shame, but not communicating that you cannot finish something will hinder all other people involved.
```
plus, here, we're talking about the point at which people have already started - so that's the difference

ykdojo commented:
^Once you address the comments I added above, I think it looks good to me.

I'll wait for one or two more reviewers to check it, though.

iShibi commented:
You should move the content inside https://github.com/ykdojo/defaang#how-to-contribute into this file and then link this file under that sub-heading.

Also, write a note for the contributors to checkout a feature branch before working on a PR instead of adding changes in the `main` branch directly.

jideabdqudus commented:
> You should move the content inside https://github.com/ykdojo/defaang#how-to-contribute into this file and then link this file under that sub-heading.
> 
> Also, write a note for the contributors to checkout a feature branch before working on a PR instead of adding changes in the `main` branch directly.

Can you give more context to what you mean here?

veganaize commented:
> > You should move the content inside https://github.com/ykdojo/defaang#how-to-contribute into this file and then link this file under that sub-heading.
> > Also, write a note for the contributors to checkout a feature branch before working on a PR instead of adding changes in the `main` branch directly.
> 
> Can you give more context to what you mean here?

@jideabdqudus I think they mean you should move the _contributing_ related stuff, from `README.md`, into `CONTRIBUTING.md`.

For their second request, about creating a new feature branch for each PR, you might reference the following document:
[https://github.com/LibreSprite/LibreSprite/blob/master/CONTRIBUTING.md#making-changes]()

jideabdqudus commented:
> > > You should move the content inside https://github.com/ykdojo/defaang#how-to-contribute into this file and then link this file under that sub-heading.
> > > Also, write a note for the contributors to checkout a feature branch before working on a PR instead of adding changes in the `main` branch directly.
> > 
> > 
> > Can you give more context to what you mean here?
> 
> @jideabdqudus I think they mean you should move the _contributing_ related stuff, from `README.md`, into `CONTRIBUTING.md`.
> 
> For their second request, about creating a new feature branch for each PR, you might reference the following document: https://github.com/LibreSprite/LibreSprite/blob/master/CONTRIBUTING.md#making-changes

I got this,

For the first request, stuff from README.md are already in here. Then I added content explaining the steps in making changes.

veganaize commented:
> > > > You should move the content inside https://github.com/ykdojo/defaang#how-to-contribute into this file and then link this file under that sub-heading.

> For the first request, stuff from README.md are already in here. Then I added content explaining the steps in making changes.

@jideabdqudus Can you remove the entire "How to contribute" section from `README.md` and place it inside of `CONTRIBUTING.md` and push that?

jideabdqudus commented:
> > > > > You should move the content inside https://github.com/ykdojo/defaang#how-to-contribute into this file and then link this file under that sub-heading.
> 
> > For the first request, stuff from README.md are already in here. Then I added content explaining the steps in making changes.
> 
> @jideabdqudus Can you remove the entire "How to contribute" section from `README.md` ~and place it inside of `CONTRIBUTING.md`~, link to `CONTRIBUTING.md` from `README.md`, and push that?

Okay that makes sense, I just did with the new push

veganaize commented:
> > @jideabdqudus Can you remove the entire "How to contribute" section from `README.md` ~and place it inside of `CONTRIBUTING.md`~, link to `CONTRIBUTING.md` from `README.md`, and push that?
> 
> Okay that makes sense, I just did with the new push

I could be wrong but it looks like you've removed the "How to contribute" section from `CONTRIBUTING.md` and not from `README.md`?  It should be the other way around.

jideabdqudus commented:
> > > @jideabdqudus Can you remove the entire "How to contribute" section from `README.md` ~and place it inside of `CONTRIBUTING.md`~, link to `CONTRIBUTING.md` from `README.md`, and push that?
> > 
> > 
> > Okay that makes sense, I just did with the new push
> 
> I could be wrong but it looks like you've removed the "How to contribute" section from `CONTRIBUTING.md` and not from `README.md`? It should be the other way around. [_Sorry if my crossed out comment confused you._]

Oh Dang, you lost me a bit. I've reverted this and fixed README.md

veganaize commented:
> Oh Dang, you lost me a bit. I've reverted this and fixed README.md

_Sorry... my bad @jideabdqudus !_

This all looks good, to me.  What do you think @ykdojo , @rohitdasu , and @iShibi ?

ykdojo reviewed (changes requested):
Can you add a link to CONTRIBUTING.md from README.md?

rohitdasu commented:
> Can you add a link to CONTRIBUTING.md from README.md?

yes agree, should have a link to CONTRIBUTING.md

jideabdqudus commented:
Okay guys, back to this :).
I've added a link to CONTRIBUTING.md from README.md with the latest push. 
Can we once go through the file so I wouldn't have to make little changes again @ykdojo @rohitdasu @iShibi

ykdojo commented:
Thank you so much! Looks good to me.

iShibi commented:
@jideabdqudus looks good. Resolve the git conflicts by rebasing and then we will merge it.

jideabdqudus commented:
> @jideabdqudus looks good. Resolve the git conflicts by rebasing and then we will merge it.

Feel free to merge now

iShibi commented:
You overwrote the changes made in the `main`. I'll have to rebase it and then we can merge.
